src/Service/Engine.php

```diff
@@ -100,12 +110,18 @@ public function dispatchEvent(string $name, DatabaseEvent $event): void
      * @param  string   $message message
      * @param  array    $context optional context data
      */
-    public function log(string $level, string $message, array $context = []): void
+    public function log(string $channel="default",string $level, string $message, array $context = []): void
     {
         if(!$this->logger) return;
 
         $context['database_root'] = (string)$this->database->getPath();
-        $this->logger->log($level, $message, $context);
+        if($channel=="default"){
+            $this->logger->log($level, $message, $context);
+        }else if($channel=="command"){
+            $this->commandLogger->log($level, $message, $context);
+        }else if($channel=="event"){
+            $this->eventLogger->log($level, $message, $context);
+        }
```

Comment:
The `Engine` is responsible for providing methods to access the different services, it really is like a service provider. It is not the `Engine`'s responsibility to care about the different channels, this violates the Single Responsibility Principle (SRP).

Instead the `DatabaseLogger` should handle this.

Also it would have been good to add an else clause that would have thrown an Exception indicating that the channel is Unsupported or Unrecognized.